src/gtfs/repository/CIFRepository.ts

```diff
@@ -79,51 +85,52 @@ export class CIFRepository {
    */
   public async getSchedules(): Promise<ScheduleResults> {
     const scheduleBuilder = new ScheduleBuilder();
-    const query = this.stream.query(`
-      SELECT 
- s.schedule_id as id,
- s.train_uid, 
- e.rsid as retail_train_id, 
- greatest(s.wef_date, COALESCE(s.import_wef_date, s.wef_date)) as runs_from, 
- least(s.weu_date, COALESCE(s.import_weu_date, s.weu_date)) as runs_to,
-
+    // If the excludeFixedLinks flag is true, we add the exclude VSTP schedules condition to the query. 
+    let excludeVstpCondition: string = this.excludeVstpSchedules == true ? "AND s.schedule_type != 'VSTP'" : "";
 
-SUBSTRING(s.valid_days, 1, 1 ) as monday,
-SUBSTRING(s.valid_days, 2, 1 ) as tuesday,
-SUBSTRING(s.valid_days, 3, 1 ) as wednesday,
-SUBSTRING(s.valid_days, 4, 1 ) as thursday,
-SUBSTRING(s.valid_days, 5, 1 ) as friday,
-SUBSTRING(s.valid_days, 6, 1 ) as saturday,
-SUBSTRING(s.valid_days, 7, 1 ) as sunday,
-
-loc.crs_code as crs_code, s.stp_indicator as stp_indicator,
-sloc.public_arrival_time, sloc.public_departure_time,
-IF(s.train_status="S", "SS", s.train_category) AS train_category, 
-IFNULL(sloc.scheduled_arrival_time, sloc.scheduled_pass_time) AS scheduled_arrival_time, 
-IFNULL(sloc.scheduled_departure_time, sloc.scheduled_pass_time) AS scheduled_departure_time,
-sloc.platform, e.atoc_code, sloc.schedule_location_id AS stop_id, 
-COALESCE(sloc.activity, "") as activity, s.reservations, s.train_class
+    const sqlQuery = `
```

Comment:
I extracted the query into a template literals because the `mysql2 query` method always add single quotes to wrap the optional condition to exclude VSTP schedule, this will cause the system throw mysql syntax error..

Reply:
I've added a comment to why I don't think the template literals are a good idea elsewhere in the review.